repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -69,25 +78,37 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
     val requestGuid = nullOrWhitespace(Option(mutableHttpRequest.getHeader(CommonHttpHeader.TRACE_GUID.toString)))
 
     val clientResponse = ((requestedTenantId, requestedContactId, requestedDeviceId) match {
-      case (None, _, _) => ResponseResult(502, "No tenant ID specified")
+      case (None, _, _) => ResponseResult(403, "No tenant ID specified")
       case (Some(tenant), _, _) if "(hybrid:.*)".r.findFirstIn(tenant).isEmpty => ResponseResult(403, "Not Authorized")
       case (_, None, _) => ResponseResult(403, "No contact ID specified")
-      case (_, _, None) => ResponseResult(502, "No device ID specified")
-      case (Some(tenant), Some(contact), Some(device)) =>
+      case (_, _, None) if !nonAuthorizedPath(mutableHttpRequest.getRequestURL.toString) => ResponseResult(403, "No device ID specified")
+      case (Some(tenant), Some(contact), device) =>
         val transformedTenant = tenant.substring(tenant.indexOf(":") + 1, tenant.length)
 
         datastoreValue(transformedTenant, contact, configuration.getValkyrieServer, requestGuid) match {
-          case deviceList: DeviceList => authorize(device, deviceList.devices, mutableHttpRequest.getMethod)
+          case deviceList: DeviceList =>
+            devicePermissions = deviceList
+            device match {
+              case Some(deviceId) => authorize(deviceId, deviceList.devices, mutableHttpRequest.getMethod)
+              case None => ResponseResult(200)
+            }
           case result: ResponseResult => result
         }
     }) match {
-      case ResponseResult(403,_) if configuration.isEnableMasking403S => ResponseResult(404, "Not Found")
+      case ResponseResult(403, _) if configuration.isEnableMasking403S => ResponseResult(404, "Not Found")
       case result => result
     }
 
     clientResponse match {
```

Comment:
Yeah, it seemed like an okay place since the comment is actually addressing the structure of the doFilter method at large. Hopefully the comment makes sense anyway?